packages/console/src/pages/UserDetails/UserSettings/UserConnections/UserSsoIdentities.tsx

```diff
@@ -0,0 +1,195 @@
+import {
+  type SsoConnectorWithProviderConfig,
+  type GetUserAllIdentitiesResponse,
+  type DesensitizedEnterpriseSsoTokenSetSecret,
+  SsoProviderType,
+} from '@logto/schemas';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import Button from '@/ds-components/Button';
+import CopyToClipboard from '@/ds-components/CopyToClipboard';
+import FormField from '@/ds-components/FormField';
+import Table from '@/ds-components/Table';
+import { type RequestError } from '@/hooks/use-api';
+import SsoConnectorLogo from '@/pages/EnterpriseSso/SsoConnectorLogo';
+
+import ConnectorTokenStatus from './ConnectorTokenStatus';
+import styles from './index.module.scss';
+
+type Props = {
+  readonly ssoIdentities?: GetUserAllIdentitiesResponse['ssoIdentities'];
+  readonly isIdentitiesLoading?: boolean;
+};
+
+type RowData = {
+  providerLogo: SsoConnectorWithProviderConfig['providerLogo'];
+  providerLogoDark: SsoConnectorWithProviderConfig['providerLogoDark'];
+  branding: SsoConnectorWithProviderConfig['branding'];
+  name: SsoConnectorWithProviderConfig['connectorName'];
+  identityId: string;
+  ssoConnectorId: string;
+  tokenStatus: {
+    isTokenStorageSupported: boolean;
+    tokenSecret?: DesensitizedEnterpriseSsoTokenSetSecret;
+  };
+};
+
+// eslint-disable-next-line unicorn/prefer-native-coercion-functions
+function isRowData(data: RowData | undefined): data is RowData {
+  return Boolean(data);
+}
```

Comment:
Why do we need the type guard here? Seems too simple to exist

Reply:
Make typescript happy